depthai_ros_driver/src/dai_nodes/sensors/rgb.cpp

```diff
@@ -0,0 +1,114 @@
+    xinControl = pipeline->create<dai::node::XLinkIn>();
+    xinControl->setStreamName(controlQName);
+    xinControl->out.link(colorCamNode->inputControl);
+}
+
+void RGB::setupQueues(std::shared_ptr<dai::Device> device) {
+    if(ph->getParam<bool>(getROSNode(), "i_publish_topic")) {
+        colorQ = device->getOutputQueue(colorQName, ph->getParam<int>(getROSNode(), "i_max_q_size"), false);
+        colorQ->addCallback(std::bind(&RGB::colorQCB, this, std::placeholders::_1, std::placeholders::_2));
+        rgbPub = image_transport::create_camera_publisher(getROSNode(), "~/" + getName() + "/image_raw");
+        if(ph->getParam<bool>(getROSNode(), "i_enable_preview")) {
+            previewQ = device->getOutputQueue(previewQName, ph->getParam<int>(getROSNode(), "i_max_q_size"), false);
+            previewQ->addCallback(std::bind(&RGB::colorQCB, this, std::placeholders::_1, std::placeholders::_2));
+            previewPub = image_transport::create_camera_publisher(getROSNode(), "~/" + getName() + "/preview/image_raw");
+        };
+        auto calibHandler = device->readCalibration();
+        rgbInfo = dai::ros::calibrationToCameraInfo(calibHandler,
+                                                    static_cast<dai::CameraBoardSocket>(ph->getParam<int>(getROSNode(), "i_board_socket_id")),
+                                                    ph->getParam<int>(getROSNode(), "i_width"),
+                                                    ph->getParam<int>(getROSNode(), "i_height"));
+    }
+    controlQ = device->getInputQueue(controlQName);
+}
+
+void RGB::closeQueues() {
+    if(ph->getParam<bool>(getROSNode(), "i_publish_topic")) {
+        colorQ->close();
+        if(ph->getParam<bool>(getROSNode(), "i_enable_preview")) {
+            previewQ->close();
+        }
+    }
+    controlQ->close();
+}
+
+void RGB::colorQCB(const std::string& name, const std::shared_ptr<dai::ADatatype>& data) {
+    auto frame = std::dynamic_pointer_cast<dai::ImgFrame>(data);
+    cv::Mat cv_frame = frame->getCvFrame();
+    auto curr_time = getROSNode()->get_clock()->now();
+    cv_bridge::CvImage img_bridge;
+    sensor_msgs::msg::Image img_msg;
+    std_msgs::msg::Header header;
+    header.stamp = curr_time;
+    header.frame_id = std::string(getROSNode()->get_name()) + "_rgb_camera_optical_frame";
+    rgbInfo.header = header;
+    img_bridge = cv_bridge::CvImage(header, sensor_msgs::image_encodings::BGR8, cv_frame);
+    img_bridge.toImageMsg(img_msg);
+    if(name == colorQName) {
+        rgbPub.publish(img_msg, rgbInfo);
+    } else {
+        previewPub.publish(img_msg, rgbInfo);
+    }
+}
+
+void RGB::link(const dai::Node::Input& in, int linkType) {
+    if(linkType == static_cast<int>(link_types::RGBLinkType::color)) {
+        colorCamNode->video.link(in);
+    } else if(linkType == static_cast<int>(link_types::RGBLinkType::preview)) {
+        colorCamNode->preview.link(in);
+    }
+}
+
+dai::Node::Input RGB::getInput(int /*linkType*/) {
```

Comment:
this is for ?

Reply:
Is this for xLinkIn ?

Reply:
I wanted to have similar interfaces in all dai_node wrappers at start (so all wrappers had to override that method), I'm fine with removing those if that's a problem